### PR TITLE
Use checksum for classname if filepath is null

### PR DESCRIPTION
**Pull request by ajaybhargavb**
https://github.com/aspnet/Razor/issues/2353

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs [hunk 1]
```diff
@@ -25,7 +27,16 @@ public class MvcViewDocumentClassifierPass : DocumentClassifierPassBase
             @namespace.Content = "AspNetCore";
 
             var filePath = codeDocument.Source.RelativePath ?? codeDocument.Source.FilePath;
-            @class.ClassName = CSharpIdentifier.GetClassNameFromPath(filePath);
+            if (string.IsNullOrEmpty(filePath))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs]: Can we capture some info here about how/why this happens? What's the possible scenarios? The code seems fine

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs [hunk 2]
```diff
@@ -37,5 +48,22 @@ public class MvcViewDocumentClassifierPass : DocumentClassifierPassBase
...
+            }
+
+            var result = new StringBuilder(bytes.Length);
+            for (var i = 0; i < bytes.Length; i++)
+            {
+                // The x2 format means lowercase hex, where each byte is a 2-character string.
+                result.Append(bytes[i].ToString("x2"));
+            }
+
+            return result.ToString();
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs]: Don't we already have this code elsewhere?

Comment (ajaybhargavb) [reply src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs]: Yup. But we don't have the IVT. So I thought if we have to duplicate in one place, I'll just duplicate it everywhere

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs (hunk 1 above)
Comment (ajaybhargavb) [reply src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs]: You mean add a comment?

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs (hunk 2 above)
Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs]: Shared file? We now have 4 copies of this code :(

**Comment by NTaylorMullen** [comment]
Part of me wishes we had a default behavior of using the checksum (assuming its not expensive) and then in MVC case if file path is available override it again.

**Comment by ajaybhargavb** [comment]
> Part of me wishes we had a default behavior of using the checksum

Maybe but there could be multiple implementations of `GetChecksum()` and some of those could be expensive. I don't think we can rely on them enough to make it the default behavior.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request closed without merging**